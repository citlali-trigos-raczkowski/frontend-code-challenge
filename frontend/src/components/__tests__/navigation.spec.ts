import { shallowMount, createLocalVue } from "@vue/test-utils";
import Navigation from "../navigation.vue";
import VueRouter from "vue-router";
import { NavigationMockData } from "../../../tests/mocks/mock-props";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("component/navigation.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(Navigation, {
      propsData: NavigationMockData,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
