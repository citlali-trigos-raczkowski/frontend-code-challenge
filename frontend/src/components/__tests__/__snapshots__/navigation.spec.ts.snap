// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/navigation.vue renders correctly 1`] = `
<div
  id="navigation"
>
  <nav-tabs-stub
    viewall="[Function]"
    viewfavorites="[Function]"
  />
   
  <div
    id="side-by-side"
  >
    <span
      id="nav-left"
    >
      <search-bar-stub
        modeoption="dark"
        updatesearchstring="[Function]"
      />
    </span>
     
    <span
      id="nav-right"
    >
      <dropdown-filter-stub
        changefilter="[Function]"
        modeoption="dark"
        pokemontypes="type1"
      />
       
      <toggle-gallery-view-stub
        galleryview="true"
        updategalleryview="[Function]"
      />
    </span>
  </div>
</div>
`;
